### PR TITLE
[chore] add list of component issues to README

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
This adds to the status README two links to issues associated with the component.

Screenshot 2023-07-08 at 11 51 06 PM

### CODE REVIEW

**Comment by TylerHelmuth** [comment]
I'll review this on Monday but I loooove this idea

**Comment by atoulme** [comment]
Filing https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/24049 for the dead link reported.

File: cmd/mdatagen/templates/readme.md.tmpl [hunk 1]
```diff
@@ -16,6 +17,9 @@
 {{- if .Status.Warnings }}
 | Warnings      | [{{ stringsJoin .Status.Warnings ", " }}](#warnings) |
 {{- end }}
+{{- if ne $class "" }}
+| Issues        | ![Open issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aopen%20label%3A{{ $class }}%2F{{ $type }}%20&label=open&color=orange&logo=opentelemetry) ![Closed issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aclosed%20label%3A{{ $class }}%2F{{ $type }}%20&label=closed&color=blue&logo=opentelemetry) |
```
Comment (dmitryax) [inline cmd/mdatagen/templates/readme.md.tmpl:21]: Label names are created from the directory names by trimming the `class` suffix while the `type` in metadata.yaml is the name of the component in the user configuration. So this link will not catch issues for the components where those two values are different, e.g., `splunk_hec`/`splunkhec`. 

We need to either update the label names to use metadata.yaml's `type` or use the directory name here similar to how it's done in https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/cmd/mdatagen/loader.go#L233

**Comment by dmitryax** [comment]
I like the idea 👍

File: cmd/mdatagen/templates/readme.md.tmpl (hunk 1 above)
Comment (atoulme) [reply cmd/mdatagen/templates/readme.md.tmpl:21]: I would update labels, I think we have a few cases but nothing too hard.

Comment (dmitryax) [reply cmd/mdatagen/templates/readme.md.tmpl:21]: I'm good with that, it just can be tricky. The bash CI automation would need to read metadata.yaml files

Comment (atoulme) [reply cmd/mdatagen/templates/readme.md.tmpl:21]: Understood, I didn’t know they were generated

Comment (atoulme) [reply cmd/mdatagen/templates/readme.md.tmpl:21]: Slept on it, and you're right. That's way too much drama, I'll use the directory name same as we did before. We will need to address this eventually maybe.

Comment (dmitryax) [reply cmd/mdatagen/templates/readme.md.tmpl:21]: > We will need to address this eventually maybe.

Right, having the `type` in the GH label eventually is preferable. Worth filing an issue

**Review by codeboten — COMMENTED** [review]
Love this idea! Thanks @atoulme

**Review by TylerHelmuth — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

File: cmd/mdatagen/templates/readme.md.tmpl (hunk 1 above)
Comment (atoulme) [reply cmd/mdatagen/templates/readme.md.tmpl]: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/24171

**Pull request merged**